jdaviz/tests/test_utils.py

```diff
@@ -14,3 +17,41 @@ def test_alpha_index_exceptions():
         utils.alpha_index(4.2)
     with pytest.raises(ValueError, match="index must be positive"):
         utils.alpha_index(-1)
+
+
+def test_uri_to_download_bad_scheme(imviz_helper):
+    uri = "file://path/to/file.fits"
+    with pytest.raises(ValueError, match=r'URI file://path/to/file\.fits with scheme file'):
+        imviz_helper.load_data(uri)
+
+
+@pytest.mark.remote_data
+def test_uri_to_download_nonexistent_mast_file(imviz_helper):
+    # this validates as a mast uri but doesn't actually exist on mast:
+    uri = "mast:JWST/product/jw00000-no-file-here.fits"
+    with pytest.raises(ValueError, match='Failed query for URI'):
+        imviz_helper.load_data(uri, cache=False)
+
+
+@pytest.mark.remote_data
+def test_url_to_download_imviz_local_path_warning(imviz_helper):
+    url = "https://www.astropy.org/astropy-data/tutorials/FITS-images/HorseHead.fits"
+    with (
+        pytest.warns(UserWarning, match='You requested to cache data'),
+        pytest.warns(FITSFixedWarning, match="'datfix' made the change")
+    ):
+        imviz_helper.load_data(url, cache=False, local_path='horsehead.fits')

```

Comment:
I don't quite understand the expected behavior of this test. If I give a direct URL, I would expect the warning to be about `local_path` being ignored because it is going through `astropy.utils.data.download_file` and not `astroquery.mast`. Instead here you are checking that the expected warning to be about `cache`. Why?

Reply:
That's actually what the warning is: 
```
    local_path_msg = (
        f"You requested to cache data to the `local_path`='{local_path}'. This "
        f"keyword argument is supported for downloads of MAST URIs via astroquery, "
        f"but since the remote file at '{possible_uri}' will be downloaded "
        f"using `astropy.utils.data.download_file`, the file will be "
        f"stored in the astropy download cache instead."
    )
```